src/Microsoft.DotNet.Compiler.Common/BindingRedirectGenerator.cs

```diff
@@ -109,26 +109,24 @@ private static XElement GetOrAddElement(XContainer parent, XName elementName)
 
         private static AssemblyRedirect[] CollectRedirects(IEnumerable<LibraryExport> dependencies)
         {
-            var allRuntimeAssemblies = dependencies.SelectMany(d => d.RuntimeAssemblies).Select(GetAssemblyInfo).ToArray();
-            var assemblyLookup = allRuntimeAssemblies.ToDictionary(r => r.Identity.ToLookupKey());
+            var dependenciesList = dependencies.ToList();
+            var allRuntimeAssemblies = dependenciesList.SelectMany(d => d.RuntimeAssemblies).Select(GetAssemblyInfo).ToArray();
+            var assemblyLookup = GetListOfUniqueRefs(dependenciesList).Select(GetAssemblyInfo).ToDictionary(r => r.Identity.ToLookupKey());
```

Comment:
You are calling GetAssemblyInfo on some assemblies twice, I'd prefer not doing this.